### ISSUE TITLE
[DISCUSSION] make a preview of new icons on release

### ISSUE THREAD

**Issue by thismoon** [LawnchairLauncher/lawnicons#1160]
### About this discussion

this is inspired by [Arcticons](https://github.com/Donnnno/Arcticons)
when releasing there should be an image previewing the new icons (or a few important ones)
the image will also contain some info like the version and number of new icons
I made one manually for 2.0.0 (using git diff, some python scripts i made and html) but it can probably be automated

![i used firefox screenshot](https://user-images.githubusercontent.com/107642606/231534432-bcc61ac9-3284-463c-8b1b-94fe2c3d34de.png)

**Comment by x9136** [comment]
It would be great to mark and filter new icons in the app itself. Something like that, between recent versions that have icon additions.
![New version additions](https://github.com/LawnchairLauncher/lawnicons/assets/60105060/a4cdcae7-0def-4d04-a826-9d55028509bf)

**Comment by thismoon** [comment]
It would be better to implement categories (e.g. calendars, clocks, social, file browsers...). I think other icon packs already do this. And one of those categories would be "Newly added" (shorter and better than "New version additions")

Also that dot for new icons feels like a notification more than a tag, it would confuse some users

And i think that [tabs](https://m3.material.io/components/tabs/overview) would be better than [chips](https://m3.material.io/components/chips/overview) unless you want to show multiple categories which will make it complicated. what do you think?

**Comment by x9136** [comment]
@thismoon

It is difficult to say which is better, so I will explain why I suggested this solution.
1. It is convenient to see what icons are new in the general list. For example, last time you were looking for some icon on C (let's say clock) and remembered by scrolling where it is approximately. So next time you can get there faster and see what's been added. Or search and see in the results marks at new icons. In my opinion a round marking won't be seen as a notification, because you're in a different context. Of course, the marking could be different because practical usefulness is more important than looks.

2. I consider tabs to be an element of navigation. In this case, I'm filtering new icons in a general list, so the tabs are wrong in meaning.

3. I didn't categorize everything for 2 reasons. The first is that everyone has different categories in their head, so to make an adequate system, you have to do [card sorting](https://www.nngroup.com/articles/card-sorting-definition/). I don't have that option. Making categories without it, people will too often guess in what category to look for the desired icon. You get titles for the sake of titles. Second: if the list consists only of new icons, then it may happen that some category will be empty or with 1-2 icons. It turns out there is little use of the output of the category here.

**Comment by thismoon** [comment]
@x9136
I agree with you on 2 and 3. But something about the round marking seems wrong (to me as a user). while some websites may use round marking on cards/buttons/other elements to indicate that it is a new feature, the user expects the marking to disappear after clicking on the element (which will not happen in our case)

> A small badge is a simple circle, used to indicate an *unread* notification.
[Badge – Material Design 3](https://m3.material.io/components/badges/guidelines)

I think it would be better to use a brighter background for new icons and when tapping on the icon there will be some text indicating that it's a new icon

**Comment by x9136** [comment]
@thismoon 
>I think it would be better to use a brighter background for new icons and when tapping on the icon there will be some text indicating that it's a new icon

When you need to explain an indication with text that means it's weak solution. A strong solution requires no explanation. Therefore, I suggest looking for a solution that doesn't need to be further explained or duplicated. For example, the connection between "New version additions" and an icon indication speaks for itself in the past version. Likewise, no explanation is needed when the indication itself says "New".

![Newly added icons](https://github.com/LawnchairLauncher/lawnicons/assets/60105060/0f72907c-027e-495f-bb57-ff0287f01014)

**Comment by x9136** [comment]
For the first version, this task can be simplified — it is enough to make 2 categories (headers):
1. New icons. In this category there will be only those icons that were not in the previous version.
2. Previously added icons. The rest of the icons will be in this category.

**Comment by Captune** [comment]
@x9136 Inspired by this, I suggest adding version categorized tabs for all versions for researching

**Comment by x9136** [comment]
> I suggest adding version categorized tabs for all versions for researching

Will this help satisfy curiosity (add convenience for this purpose) or is there some other benefit?

**Comment by Captune** [comment]
> @Captune  Will this help satisfy curiosity (add convenience for this purpose) or is there some other benefit?

🤔 help satisfy curiosity, it also gives clarity, makes version differences more discoverable, maybe gives contributors a sense of accomplishment

**Comment by x9136** [comment]
You can view the new icons in a separate category on the Lawnicons home screen.
